renku/core/utils/git.py

```diff
@@ -136,6 +138,43 @@ def get_object_hash_from_submodules() -> Optional[str]:
         return get_object_hash_from_submodules()
 
 
+def get_content(repo: Repo, path: Union[Path, str], checksum: str) -> str:
+    """Get content of an object from git using its checksum and return a path with that content."""
+
+    def get_content_from_submodules(file):
+        for submodule in repo.submodules:
+            try:
+                path_in_submodule = Path(path).relative_to(submodule.path)
+            except ValueError:
+                continue
+            else:
+                try:
+                    subprocess.run(
+                        ["git", "cat-file", "--filters", "--path", str(path_in_submodule), checksum],
+                        check=True,
+                        stdout=file,
+                        cwd=repo.working_dir,
+                    )
+                except subprocess.CalledProcessError:
+                    pass
+                else:
+                    return True
+
+    with tempfile.NamedTemporaryFile(mode="w+b", delete=False) as temp_file:
+        try:
+            subprocess.run(
+                ["git", "cat-file", "--filters", "--path", str(path), checksum],
```

Comment:
Do we need `-e` flag here? 
```
Exit with zero status if <object> exists and is a valid object. If <object> is of an invalid format exit with non-zero and emits an error on stderr.
```

Also what does this do with LFS files? And how does it handle symlinks?

Reply:
`-e` only checks if the object exits and won't print its content. `--filters` takes care of LFS and complains if object is invalid.
This won't work with symlinks and I couldn't find a workaround for it. However, we don't have symlinks in the project except for external files. AFAIK, we don't have a clear policy on how to handle external files on export/import. So, I'll create a story for this.